common/src/stack/command/stack/commands/list/storage/partition/__init__.py

```diff
@@ -8,148 +8,110 @@
-					nodes as a on p.tableid=a.id where p.scope='host') UNION ALL
-					(select a.name, p.device, p.mountpoint, p.size,
-					p.fstype, p.options, p.partid from storage_partition as p inner join
-					appliances as a on p.tableid=a.id where
-					p.scope='appliance') order by scope,device,partid,size,fstype"""
-		elif scope == 'os':
-			query = """select scope, device, mountpoint, size, fstype, options, partid
-				from storage_partition where scope = "os" and tableid = (select id
-				from oses where name = %s) order by device,partid,fstype,size"""
-		elif scope == 'appliance':
-			query = """select scope, device, mountpoint, size, fstype, options, partid
-				from storage_partition where scope = "appliance"
-				and tableid = (select id from appliances
-				where name = %s) order by device,partid,fstype, size"""
-		elif scope == 'host':
-			query = """select scope, device, mountpoint, size, fstype, options, partid
-				from storage_partition where scope="host" and
-				tableid = (select id from nodes
-				where name = %s) order by device,partid,fstype, size"""
-
-		if not query:
-			return
+	def run(self, params, args):
+		# Get the scope and make sure the args are valid
+		scope, = self.fillParams([('scope', 'global')])
+		scope_mappings = self.getScopeMappings(args, scope)
 
 		self.beginOutput()
-
-		if scope == 'global':
-			self.db.execute(query)
+		for scope_mapping in scope_mappings:
+			if scope == 'host':
+				# Get the host's info for the scope linking
+				host, appliance_id, os_id, environment_id = self.db.select("""
+					nodes.name, appliance, boxes.os, environment
+					FROM nodes, boxes
+					WHERE nodes.id = %s AND nodes.box = boxes.id
+				""", (scope_mapping.node_id,))[0]
+
+				# Get the controller data for all scopes that match the host's info
+				rows = self.db.select("""
+					storage_partition.device, storage_partition.partid,
+					storage_partition.mountpoint, storage_partition.size,
+					storage_partition.fstype, storage_partition.options,
+					UPPER(LEFT(scope_map.scope, 1))
+					FROM storage_partition
+					INNER JOIN scope_map
+						ON storage_partition.scope_map_id = scope_map.id
+					WHERE scope_map.scope = 'global'
+					OR (scope_map.scope = 'appliance' AND scope_map.appliance_id <=> %s)
+					OR (scope_map.scope = 'os' AND scope_map.os_id <=> %s)
+					OR (scope_map.scope = 'environment' AND scope_map.environment_id <=> %s)
+					OR (scope_map.scope = 'host' AND scope_map.node_id <=> %s)
+					ORDER BY scope_map.scope DESC, device, partid, size, fstype
+				""", (appliance_id, os_id, environment_id, scope_mapping.node_id))
+
+				# The routes come out of the DB with the higher value scopes
+				# first. First scope wins, so we output until the scope changes.
+				last_scope = None
+				for row in rows:
+					if last_scope and last_scope != row[6]:
+						break
+
+					self.addOutput(host, self._clean_data(*row))
+
+					last_scope = row[6]
+			else:
```

Comment:
I was trying out the new scopes and noticed with my cluster up system with one backend, the global scope of `list storage partition` outputs nothing, when I'm assuming it's supposed to output every partition config in the database. I made the required changes to the db as well so the appliance and host scopes are working. Not sure if that is the right behavior for the command with only a backend partition config but maybe the data here isn't being output to the command?